src/instajam.js

```diff
@@ -213,11 +213,19 @@
       // Make a request to the API
       User.prototype.search.call(this, id, {}, function(result) {
 
-        // If the initial user search yields any
-        // results, then just return the first, but
+        // Go through the results and check for
+        // a perfect match, then query again with that user's ID
         // otherwise return nothing.
-        if (result.data && result.data.length === 1) {
-          result = result.data[0];
+        if (result.data) {    
+            var userIds = result.data;
+            result = false;
+
+            for (var i = 0; i < userIds.length; i++) {
+                if (userIds[i].username == id) {
+                    User.prototype.get(parseInt(userIds[i].id), callback);
```

Comment:
Please use a radix for `parseInt`, a la `parseInt(userIds[i].id, 10)`. See [MDN for an explanation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt?redirectlocale=en-US&redirectslug=JavaScript%2FReference%2FGlobal_Objects%2FparseInt).

>  For this reason always specify a radix when using parseInt.